pkg/csi/service/common/commonco/k8sorchestrator/k8sorchestrator.go

```diff
@@ -864,11 +864,12 @@ func pvAdded(obj interface{}) {
 	// Add VCP-CSI migrated volumes to the volumeIDToNameMap map.
 	// Since cns query will return all the volumes including the migrated ones, the map would need to be a
 	// union of migrated VCP-CSI volumes and CSI volumes, as well.
-	if pv.Spec.CSI == nil && (k8sOrchestratorInstance.IsFSSEnabled(context.Background(), common.CSIMigration) &&
-		pv.Spec.VsphereVolume != nil && isValidMigratedvSphereVolume(context.Background(), pv.ObjectMeta)) {
+	if pv.Spec.VsphereVolume != nil &&
+		k8sOrchestratorInstance.IsFSSEnabled(context.Background(), common.CSIMigration) &&
+		isValidMigratedvSphereVolume(context.Background(), pv.ObjectMeta) {
 		if pv.Status.Phase == v1.VolumeBound {
-			k8sOrchestratorInstance.volumeIDToNameMap.add(pv.Spec.CSI.VolumeHandle, pv.Name)
-			log.Debugf("Migrated pvAdded: Added '%s -> %s' pair to volumeIDToNameMap", pv.Spec.CSI.VolumeHandle, pv.Name)
+			k8sOrchestratorInstance.volumeIDToNameMap.add(pv.Spec.VsphereVolume.VolumePath, pv.Name)
```

Comment:
Here we are adding volumePath as the key. Can you confirm if volumeIDToNameMap.get(volumePath) is used accordingly

Reply:
yes, we are adding a path for migrated volume, because we want `GetAllK8sVolumes` to return both CSI and Migrated volumes. It is only used for ListVolumeThreshold check.

```
volIDsInK8s = commonco.ContainerOrchestratorUtility.GetAllK8sVolumes()
.
.

if len(volIDsInK8s)-len(CNSVolumesforListVolume) > cfg.Global.ListVolumeThreshold {
```

Reply:
we are using this internally in `GetNodesForVolumes` to retrieve volumename for ID
but it is used only for file volume in vanilla for `processQueryResultsListVolumes` function.

```
publishedNodeIds := commonco.ContainerOrchestratorUtility.GetNodesForVolumes(ctx, []string{fileVolID})
```

There is no impact on WCP for this change.

Reply:
Right, but the map name "volumeIDToNameMap" does not convey that it includes (volumePath || volumeID).

May be we can add some comments to the map definition to indicate that the key can be volumeID or volumePath.

Reply:
> we are using this internally in `GetNodesForVolumes` to retrieve volumename for ID
> but it is used only for file volume in vanilla for `processQueryResultsListVolumes` function.

On a seconds thought, do we even have to add migrated volumes to the map. If the map.get() is only for fileVolumes. Let me know your thoughts ?

Reply:
> Right, but the map name "volumeIDToNameMap" does not convey that it includes (volumePath || volumeID).

Let me put some comment to convey that.

> On a seconds thought, do we even have to add migrated volumes to the map, since map.get() is only for fileVolumes. Let me know your thoughts ?

We need to add just for honoring `if len(volIDsInK8s)-len(CNSVolumesforListVolume) > cfg.Global.ListVolumeThreshold ` check. If we do not add then, CNSVolumesforListVolume length will be always greater than volIDsInK8s which is retrieved using `commonco.ContainerOrchestratorUtility.GetAllK8sVolumes()`.

code is not organized properly, we definitely need to revisit this and clean it up.